ComPact/Models/V3/Request.cs

```diff
@@ -101,15 +102,32 @@ public bool Match(Request actualRequest)
             var pathsMatch = Path == actualRequest.Path;
             var headersMatch = Headers.Match(actualRequest.Headers);
             var queriesMatch = Query.Match(actualRequest.Query);
-            var bodiesMatch = Body == actualRequest.Body;
-
+            var bodiesMatch = MatchBody(actualRequest);
+            
             return methodsMatch && pathsMatch && headersMatch && queriesMatch && bodiesMatch;
         }
 
         internal void SetEmptyValuesToNull()
         {
             Headers = Headers.Any() ? Headers : null;
             Query = Query.Any() ? Query : null;
+        }
+
+        internal bool MatchBody(Request actualRequest)
+        {
+            bool bodiesMatch = false;
+
+            if (Body is JObject && actualRequest.Body is JObject)

```

Comment:
Now it only works when the body is literally a JObject. Have you considered just parsing the body to a JToken anyway?
`var bodiesMatch = JObject.DeepEquals(JToken.FromObject(Body), JToken.FromObject(actualRequest.Body));` seems to work.

Reply:
Updated.  (didn't want to make large changes)